scvi/metrics/classification.py

```diff
@@ -84,29 +86,35 @@ def compute_accuracy_svc(data_train, labels_train, data_test, labels_test, unit_
     svc = SVC(max_iter=max_iter)
 
     clf = GridSearchCV(svc, param_grid, verbose=verbose)
-    clf.fit(data_train, labels_train)
-
-    # Predicting the labels
-    y_pred_test = clf.predict(data_test)
-    y_pred_train = clf.predict(data_train)
-
-    return (compute_accuracy_tuple(labels_train, y_pred_train),
-            compute_accuracy_tuple(labels_test, y_pred_test))
+    return compute_accuracy_classifier(clf, data_train, labels_train, data_test, labels_test, return_labels)
 
 
-def compute_accuracy_rf(data_train, labels_train, data_test, labels_test, unit_test=False, verbose=0):
+def compute_accuracy_rf(data_train, labels_train, data_test, labels_test, unit_test=False,
+                        verbose=0, return_labels="False"):
     param_grid = {'max_depth': np.arange(3, 10), 'n_estimators': [10, 50, 100, 200]}
     if unit_test:
         param_grid = [{'max_depth': [3], 'n_estimators': [10]}]
 
     rf = RandomForestClassifier(max_depth=2, random_state=0)
 
     clf = GridSearchCV(rf, param_grid, verbose=verbose)
-    clf.fit(data_train, labels_train)
+    return compute_accuracy_classifier(clf, data_train, labels_train, data_test, labels_test, return_labels)
+
+
+def compute_accuracy_nn(data_train, labels_train, data_test, labels_test, k=5, return_labels="False"):
+    clf = neighbors.KNeighborsClassifier(k, weights='distance')
+    return compute_accuracy_classifier(clf, data_train, labels_train, data_test, labels_test, return_labels)
 
+
+def compute_accuracy_classifier(clf, data_train, labels_train, data_test, labels_test, return_labels=False):
+    clf.fit(data_train, labels_train)
     # Predicting the labels
     y_pred_test = clf.predict(data_test)
     y_pred_train = clf.predict(data_train)
 
-    return (compute_accuracy_tuple(y_pred_train, labels_train),
-            compute_accuracy_tuple(y_pred_test, labels_test))
+    if not return_labels:
+        return (compute_accuracy_tuple(labels_train, y_pred_train),
+                compute_accuracy_tuple(labels_test, y_pred_test))
+    elif return_labels:
+        return (compute_accuracy_tuple(labels_train, y_pred_train),
+                compute_accuracy_tuple(labels_test, y_pred_test), y_pred_test)
```

Comment:
Would you just always return the labels, rather than having this if statement? If you call the function and you don't need them, you can just use and underscore as a variable to assign to; e.g.,
```
train_acc, test_acc, _ = infer.compute_accuracy_classifier(...)
```

Reply:
I'll do that and modify accordingly the part of the code where we call that function.